src/main/java/catan/services/util/play/MainStageUtil.java

```diff
@@ -30,6 +36,45 @@ public void updateNextMove(GameBean game) {
         game.setCurrentMove(nextMoveNumber);
     }
 
+    public void distributeResources(GameBean game) {
+        Integer diceValue = game.getDiceFirstValue() != null ? game.getDiceFirstValue() + game.getDiceSecondValue() : 0;
+        for (HexBean hex : game.getHexes()) {
+            if (diceValue.equals(hex.getDice())) {
+                for (NodeBean node : hex.getNodes().listAllNotNullItems()) {
+                    if (node.getBuilding() != null) {
+                        giveResourceToPlayer(node.getBuilding().getBuildingOwner(), node.getBuilding().getBuilt(), hex.getResourceType());
+                    }
```

Comment:
I see you like to have deep nesting logic) there are 6 tabs between border and giveResourceToPlayer() method call, it should tell you that this code is not readable